pkg/compact/compact.go

```diff
@@ -966,16 +971,28 @@ func (c *BucketCompactor) Compact(ctx context.Context) error {
 		}
 
 		// Send all groups found during this pass to the compaction workers.
+	groupLoop:
 		for _, g := range groups {
 			select {
-			case <-errGroupCtx.Done():
-				break
+			case err = <-errChan:
+				if err != nil {
+					break groupLoop
+				}
 			case groupChan <- g:
 			}
 		}
 		close(groupChan)
-		if err := errGroup.Wait(); err != nil {
-			return err
+		wg.Wait()
+
+		close(errChan)
+		workCtxCancel()
```

Comment:
Why you cancel if all go routines already left? just to release resources right?

Reply:
Yeah just to release the resources